cpp/demo/interpolation/main.cpp

```diff
@@ -0,0 +1,73 @@
+#include <algorithm>
+#include <cmath>
+
+#include <dolfinx.h>
+#include <dolfinx/io/XDMFFile.h>
+
+#include "interpolation.h"
+
+using namespace dolfinx;
+
+int main(int argc, char *argv[])
+{
+  common::subsystem::init_logging(argc, argv);
+  common::subsystem::init_petsc(argc, argv);
+
+  // Set the logging thread name to show the process rank
+  int mpi_rank;
+  MPI_Comm_rank(MPI_COMM_WORLD, &mpi_rank);
+  std::string thread_name = "RANK " + std::to_string(mpi_rank);
+  loguru::set_thread_name(thread_name.c_str());
+  if (dolfinx::MPI::rank(MPI_COMM_WORLD) == 0) {
+      loguru::g_stderr_verbosity = loguru::Verbosity_INFO;
+  }
+
+  {
+    auto meshL = std::make_shared<mesh::Mesh>(generation::BoxMesh::create(
+        MPI_COMM_WORLD, {{{0.0, 0.0, 0.0}, {1.0, 1.0, 1.0}}}, {16, 16, 16},
+        mesh::CellType::tetrahedron, mesh::GhostMode::shared_facet));
+
+    auto meshR = std::make_shared<mesh::Mesh>(generation::BoxMesh::create(
+        MPI_COMM_WORLD, {{{0.25, 0.25, 0.25}, {0.75, 0.75, 0.75}}},
+        {16, 16, 16}, mesh::CellType::tetrahedron,
+        mesh::GhostMode::shared_facet));
+
+    auto VL = fem::create_functionspace(functionspace_form_interpolation_a, "u",
+                                        meshL);
+    auto VR = fem::create_functionspace(functionspace_form_interpolation_a, "u",
+                                        meshR);
+
+    auto uL = std::make_shared<fem::Function<PetscScalar>>(VL);
+    auto uR = std::make_shared<fem::Function<PetscScalar>>(VR);
+    auto f = std::make_shared<fem::Constant<PetscScalar>>(1);
+
+    uL->interpolate([](auto& x) {
+      xt::xtensor<PetscScalar, 1> r(xt::shape({x.shape()[1]}), 0);
+      for (std::size_t i = 0; i < x.shape(1); ++i)
+      {
+        r(0, i) = std::cos(10 * x(0, i)) * std::sin(10 * x(2, i));
+        //              r(0, i) = x(1, i);
+      }
+      return r;
+    });
+    // Is this needed?
+    VecGhostUpdateBegin(uL->vector(), INSERT_VALUES, SCATTER_FORWARD);
```

Comment:
This should not be needed, and if needed you should rather use: https://github.com/FEniCS/dolfinx/blob/d933045f09153c2e1f88682fe562440131f3f94c/cpp/dolfinx/la/Vector.h#L127-L131

Reply:
Thanks, this was again written before the new API appeared. I changed it, even though this test was not really meant to stay.